src/ring_theory/discrete_valuation_ring.lean

```diff
@@ -174,18 +174,16 @@ end
 
 /-- Implementation detail: an integral domain in which there is a unit `p`
 such that every nonzero element is associated to a power of `p` is a unique factorization domain.
-
 See `discrete_valuation_ring.of_has_unit_mul_pow_irreducible_factorization`. -/
-noncomputable def ufd : unique_factorization_domain R :=
+theorem ufd : unique_factorization_monoid R :=
```

Comment:
Should this become an instance, now that `unique_factorization_monoid` is a `Prop` and all defeq issues are history?

Reply:
The problem is that it has a (sort of hidden) hypothesis variable, so it's an impossible instance, according to the linter.